public/app/features/logs/components/log-context/LogRowContextModal.tsx

```diff
@@ -260,22 +293,108 @@ export const LogRowContextModal: React.FunctionComponent<LogRowContextModalProps
     }
   };
 
-  useLayoutEffect(() => {
-    if (!loading && entryElement.current && preEntryElement.current) {
-      preEntryElement.current.scrollIntoView({ block: 'center' });
-      entryElement.current.scrollIntoView({ block: 'center' });
+  const maybeLoadMore = async (place: Place) => {
+    const section = context[place];
+    if (section.loadingState === LoadingState.Loading) {
+      return;
+    }
+
+    setSection(place, (s) => ({
+      ...s,
+      loadingState: LoadingState.Loading,
+    }));
+
+    const currentGen = generationRef.current;
+    try {
+      const newRows = await loadMore(place);
+      if (currentGen === generationRef.current) {
+        setSection(place, (s) => ({
+          rows: place === 'above' ? [...newRows, ...s.rows] : [...s.rows, ...newRows],
+          loadingState: newRows.length === 0 ? LoadingState.Done : LoadingState.NotStarted,
+        }));
+      } else {
+        // do nothing, we were told to ignore the result of this fetch
+      }
```

Comment:
I understand the comment, but do we need the empty else just for that?

Reply:
good point, `else` removed.